cluster/images/canton/Dockerfile

```diff
@@ -1,22 +1,25 @@
 # Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-# xx provides tools to support easy cross-compilation from Dockerfiles, see https://github.com/tonistiigi/xx
-FROM --platform=$BUILDPLATFORM tonistiigi/xx@sha256:923441d7c25f1e2eb5789f82d987693c47b8ed987c4ab3b075d6ed2b5d6779a3 AS xx
-ARG BUILDPLATFORM
-ARG TARGETPLATFORM
+ARG BASE_IMAGE="eclipse-temurin:21-jdk-noble@sha256:5a65f334da5a91a66076735d78e3ae30483a2593ac108f830dcd59521f2535cd"
 
-FROM --platform=$BUILDPLATFORM debian:bookworm-slim@sha256:6ac2c08566499cc2415926653cf2ed7c3aedac445675a013cc09469c9e118fdd AS build
+FROM fullstorydev/grpcurl:v1.9.2@sha256:0c3796edc8b6bb1814da2f11a554996237c0dd98380fce182efb1dc44c556ac1 AS grpcurl
+
+FROM ${BASE_IMAGE}
```

Comment:
This currently fails CI due to
```
cluster/images/canton/Dockerfile:8:FROM ${BASE_IMAGE}
ERROR: docker images must pin base images to a specific digest
```

Do we need the indirection through the arg or can we just inline it to make CI happy?

Reply:
> Do we need the indirection through the arg or can we just inline it to make CI happy?

In this case we would have to specify and maintain the same image/hash twice, one in FROM and one in LABEL. I've updated the check script to make it parse Dockerfile instead.